src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -742,6 +744,13 @@ static int enumerate_partitions(dev_t devnum) {
         if (r < 0)
                 return log_debug_errno(r, "Failed to open %s: %m", devname);
 
+        xsprintf_sys_block_path(p, "/size", devnum);
+        r = read_one_line_file(p, &s);
+        if (r == 0 && !strcmp(s, "0")) {
```

Comment:
`read_one_line_file()` returns the number of characters read. If the file is empty, this will be 0. The conditional, as written, cannot be true.

Did this get tested at all?